openedx/adg/lms/webinars/views.py

```diff
@@ -0,0 +1,53 @@
+"""
+All views for webinars app
+"""
+from django.contrib.auth.decorators import login_required
+from django.http import HttpResponseServerError
+from django.shortcuts import get_object_or_404, redirect, render
+from django.urls import reverse
+from django.utils.translation import ugettext as _
+from django.views.decorators.csrf import ensure_csrf_cookie
+from django.views.decorators.http import require_POST
+
+from .helpers import send_webinar_registration_email
+from .models import Webinar, WebinarRegistration
+
+
+def webinar_description_page_view(request, pk):
+    # TODO This is a temp view and will be properly functional in LP-2623
+    return render(request, 'adg/lms/webinar/description_page.html', context={'pk': pk})
+
+
+@require_POST
+@login_required
+@ensure_csrf_cookie
+def webinar_registration_view(request, pk, action):
+    """
+    A view to register user in a webinar event or cancel already registered user in a webinar.
+    Args:
+        request (HttpRequest): HTTP request
+        pk (int): Webinar id, the primary key of model
+        action (str): Actions to perform on event i.e. `register` or `cancel`
+    Returns:
+        HttpResponse: On success redirect to webinar page otherwise through HTTP error code as per nature of error
+        i.e. 404 or 500
+    """
+    webinar = get_object_or_404(Webinar, pk=pk)
+    is_registering = action == 'register'
+    user = request.user
+
+    if is_registering and webinar.status == Webinar.CANCELLED:
+        return HttpResponseServerError(_('You cannot register cancelled event'))
+
+    registered_webinar_for_user = WebinarRegistration.objects.filter(
+        webinar=webinar, user=user, is_registered=is_registering
+    ).first()
+
+    if not registered_webinar_for_user:
+        WebinarRegistration.objects.update_or_create(
+            webinar=webinar, user=user, defaults={'is_registered': is_registering}
+        )
+        # pylint: disable=expression-not-assigned
+        send_webinar_registration_email(webinar, user.email) if is_registering else None
+
+    return redirect(reverse('webinar_event', kwargs={'pk': pk}))
```

Comment:
Can we please use a Class based view instead, as we have been using CBV over FBV previously.
Also, it makes the code cleaner and more extensible.

Also, kindly take a look if we can utilize a Generic CBV here.